sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -45,6 +45,70 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
     withOrigin(ctx)(StructType(visitColTypeList(ctx.colTypeList)))
   }
 
+  override def visitSingleRoutineParamList(ctx: SingleRoutineParamListContext): StructType = {
```

Comment:
where do we fail this method if unsupported feature such as generated column is specified?

Reply:
It's checked in SparkSqlParser. I should add comment in visitStructField to explain

Reply:
done